specification/batch/Azure.Batch/routes.tsp

```diff
@@ -0,0 +1,2154 @@
+
+  #deprecated "Warning: This operation is deprecated and will be removed after February, 2024. Please use the [Azure KeyVault Extension](https://learn.microsoft.com/azure/batch/batch-certificate-migration-guide) instead."
+  @summary("Creates a Certificate to the specified Account.")
+  @doc("Creates a Certificate to the specified Account.")
+  @route("/certificates")
+  @projectedName("java", "createCertificateInternal")
+  createCertificate is Ops.CreateOperation<
+    {
+      @doc("The Certificate to be created.")
+      @body
+      certificate: BatchCertificate;
+    },
+    {}
+  >;
+
+  #deprecated "Warning: This operation is deprecated and will be removed after February, 2024. Please use the [Azure KeyVault Extension](https://learn.microsoft.com/azure/batch/batch-certificate-migration-guide) instead."
+  @summary("Lists all of the Certificates that have been added to the specified Account.")
+  @doc("Lists all of the Certificates that have been added to the specified Account.")
+  @route("/certificates")
+  @projectedName("java", "listCertificatesInternal")
+  listCertificates is Ops.ListOperation<
+    {
+      @doc("""
+An OData $filter clause. For more information on constructing this filter, see
+https://docs.microsoft.com/en-us/rest/api/batchservice/odata-filters-in-batch#list-certificates.
+""")
+      @query
+      @projectedName("client", "filter")
+      $filter?: string;
+
+      @doc("An OData $select clause.")
+      @query({
+        format: "csv",
+      })
+      @projectedName("client", "select")
+      $select?: string[];
+    },
+    BatchCertificateListResult
+  >;
+
+  #deprecated "Warning: This operation is deprecated and will be removed after February, 2024. Please use the [Azure KeyVault Extension](https://learn.microsoft.com/azure/batch/batch-certificate-migration-guide) instead."
+  @summary("Cancels a failed deletion of a Certificate from the specified Account.")
+  @doc("""
+If you try to delete a Certificate that is being used by a Pool or Compute
+Node, the status of the Certificate changes to deleteFailed. If you decide that
+you want to continue using the Certificate, you can use this operation to set
+the status of the Certificate back to active. If you intend to delete the
+Certificate, you do not need to run this operation after the deletion failed.
+You must make sure that the Certificate is not being used by any resources, and
+then you can try again to delete the Certificate.
+""")
+  @route("/certificates(thumbprintAlgorithm={thumbprintAlgorithm},thumbprint={thumbprint})/canceldelete")
+  @projectedName("java", "cancelCertificateDeletionInternal")
+  cancelCertificateDeletion is Ops.PostOperation<
+    {
+      @doc("The algorithm used to derive the thumbprint parameter. This must be sha1.")
+      @path
+      thumbprintAlgorithm: string;
+
+      @doc("The thumbprint of the Certificate being deleted.")
+      @path
+      thumbprint: string;
+    },
+    NoContentSuccesResponseHeaders
+  >;
```

Comment:
Are we sure the `@route` is correct? (I cannot say I understand what it means...)

I got compiler error if run local
```
C:/github/azure-sdk-for-java/sdk/batch/azure-compute-batch/TempTypeSpecFiles/Azure.Batch/routes.tsp:862:3 - error @typespec/http/missing-path-param: Path contains parameter thumbprintAlgorithm},thumbprint={thumbprint but wasn't found in given parameters
> 862 |   cancelCertificateDeletion is Ops.PostOperation<
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^
C:/github/azure-sdk-for-java/sdk/batch/azure-compute-batch/TempTypeSpecFiles/Azure.Batch/routes.tsp:890:3 - error @typespec/http/missing-path-param: Path contains parameter thumbprintAlgorithm},thumbprint={thumbprint but wasn't found in given parameters
> 890 |   deleteCertificate is Ops.DeleteOperation<
      |   ^^^^^^^^^^^^^^^^^
C:/github/azure-sdk-for-java/sdk/batch/azure-compute-batch/TempTypeSpecFiles/Azure.Batch/routes.tsp:911:3 - error @typespec/http/missing-path-param: Path contains parameter thumbprintAlgorithm},thumbprint={thumbprint but wasn't found in given parameters
> 911 |   getCertificate is Ops.ReadOperation<
      |   ^^^^^^^^^^^^^^
```

compiler seems taking `{thumbprintAlgorithm},thumbprint={thumbprint}` as the segment param.

Reply:
this has been fixed